sql-native/src/main/java/org/ballerinalang/sql/datasource/SQLDatasource.java

```diff
@@ -135,28 +155,109 @@ public static SQLDatasource retrieveDatasource(SQLDatasource.SQLDatasourceParams
         return sqlDatasourceToBeReturned;
     }
 
+    public static Connection getConnection(TransactionResourceManager trxResourceManager, BObject client,
+            SQLDatasource datasource)
+            throws SQLException {
+        Connection conn;
+        try {
+            boolean isInTransaction = trxResourceManager.isInTransaction();
+            if (!isInTransaction) {
+                conn = datasource.getConnection();
+                return conn;
+            } else {
+                //This is when there is an infected transaction block. But this is not participated to the transaction
+                //since the action call is outside of the transaction block.
+                if (!trxResourceManager.getCurrentTransactionContext().hasTransactionBlock()) {
+                    conn = datasource.getConnection();
+                    return conn;
```

Comment:
```suggestion
                    return datasource.getConnection();
```

Reply:
Will update with a commit

Reply:
Changed. commit id is 24c39ff738c064c229053d918be3e4d986350b6a